contents/tutorials/array-filter-breakdown.md

```diff
@@ -0,0 +1,88 @@
+
+- Select a property that is already an array like `properties.$active_feature_flags`.
+
+- Split a string with `splitByChar`.  For example, splitting the [autocapture element chain](/tutorials/hogql-autocapture) with`splitByChar('.', elements_chain)`.
+
+- Extract an array from JSON with `JSONExtractArrayRaw` or `JSONExtractKeys`.
+
+Once you can access the array, you can breakdown or filter using it.
+
+## Breaking down arrays
+
+Once we have our array, we can break it down further. To do this, [create an insight](https://app.posthog.com/insights/new), click "Add breakdown" under "Breakdown by," and select HogQL. Here we can write our HogQL expressions to break down arrays. 
+
+The most useful breakdown with arrays is `arrayJoin`. This is a [special expression](https://clickhouse.com/docs/en/sql-reference/functions/array-join) that "unfolds" an array into multiple rows. and helps us access each value from within an array. For example, to get a count of the usage of different `$active_feature_flags`, you can use this HogQL expression breakdown:
+
+```sql
+arrayJoin(
+  JSONExtractArrayRaw(
+    properties.$active_feature_flags ?? '[]'
+  )
+)
+```
+
+> Adding `?? '[]'`  is a "nullish coalescing operator" that provides a default value to your properties. In this case, it provides an empty array if `properties.$active_feature_flags` is null or undefined.
+
+![Breakdown HogQL](../images/tutorials/array-filter-breakdown/breakdown.png)
+
+More useful array breakdowns include:
+
+- `arrayUniq` for counting the number of unique values in the array.
+- `arrayDistinct` which returns an array containing only the distinct elements. You can combine this with `arrayJoin` to get counts for distinct values within arrays.
+- `arrayElement(arr, 1)` for breaking down the first elements in the array.
+- `arrayElement(arr, -1)` for breaking down the last elements in the array.
+
+Once you get individual number or string values from the breakdown, you can modify these values further too. Read our [HogQL expression docs](/docs/hogql/expressions) for more useful functions for doing this.
+
+## Filtering arrays
+
+Once you break down your array, you might find it contains values you don't care about. You can use filters to remove them. Filtering arrays is relatively similar to breaking down, except you must provide values you want to filter out.
+
+To add a filter, click the filter dropdown next to your data series, click "Add filter," select HogQL, and add your expression.
+
+To start, you can remove empty arrays with a `notEmpty()` check. For example, to remove empty arrays from the `$active_feature_flags` by filtering for the HogQL expression:
+
+```sql
+notEmpty(
+  JSONExtractArrayRaw(
+    properties.$active_feature_flags ?? '[]'
+  )
+)
+```
+
+![Filter HogQL](../images/tutorials/array-filter-breakdown/filter.png)
+
+If you want to filter arrays for specific values, you can use `has`, `hasAll`, and `hasAny`:
+
```

Comment:
I also noticed when filtering for numbers, like an array like this: `[1, 2, 3]`, we still have to use a string in the filter: `has(JSONExtractArrayRaw(assumeNotNull(properties.eee.test)), '3')`. Might be worth adding a warning.